scripts/smart-account-factory-addstake.ts

```diff
@@ -0,0 +1,64 @@
+import { ethers } from "hardhat";
+import { SmartAccountFactory__factory } from "../typechain";
+import { factoryStakeConfig, isContract } from "./utils";
+import { EntryPoint__factory } from "@account-abstraction/contracts";
+import { formatEther } from "ethers/lib/utils";
+
+const entryPointAddress =
+  process.env.ENTRY_POINT_ADDRESS ||
+  "0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789";
+const smartAccountFactoryAddress = "0x9A9e92dA02485158bf4AaAA7C59398048B1cEB2D";
+
+export async function addStake() {
```

Comment:
maybe verify the owner with account doing these transactions. if passes proceed and at the end transfer ownership to rightful owner if needed.

Reply:
also we can get factory address from the salt.

Reply:
> also we can get the factory address from the salt.
I want to keep the staking script independent of the deployment scripts since there could be a situation where the salts in the config correspond to a different deployment, and we wish to stake a different factory

Reply:
cool. we can keep it as it is. in paymaster repo PR, staking script I noticed we used token paymaster address from the salt.